lib/temporal/testing/replay_tester.rb

```diff
@@ -0,0 +1,136 @@
+require 'gen/temporal/api/history/v1/message_pb'
+require 'json'
+require 'temporal/errors'
+require 'temporal/metadata/workflow_task'
+require 'temporal/middleware/chain'
+require 'temporal/workflow/executor'
+require 'temporal/workflow/stack_trace_tracker'
+
+module Temporal
+  module Testing
+    class ReplayError < StandardError
+    end
+
+    class ReplayTester
+      def initialize(config: Temporal.configuration)
+        # Duplicate the configuration so that this doesn't interfere with other tests in the
+        # same process that are not replay tests
+        @config = config.dup.tap do |c|
+          # Otherwise, replay tests will produce no logs. This helps with test debugging.
```

Comment:
Interesting idea.  I like it, and I couldn't find evidence that the temporal SDKs do this.
I think people would want this in tests, but perhaps not if doing safe-rollout deploys (might be scary to see redundant logs during production rollouts).  I wonder if this should be configurable.

Reply:
Could be a nice option to have during testing, certainly, but yeah I'd make it configurable defaulting to off.